test/e2e/agent/agent_test.go

```diff
@@ -328,6 +328,45 @@ var _ = Describe("VM Agent behavior", func() {
 
 		})
 
+		It("K8s secret config source", func() {
+			deviceId, _ := harness.EnrollAndWaitForOnlineStatus()
+
+			// Get the next expected rendered version
+			newRenderedVersion, err := harness.PrepareNextDeviceVersion(deviceId)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("should report the correct device status after an inline config is added")
+
+			err = harness.UpdateDeviceWithRetries(deviceId, func(device *v1alpha1.Device) {
+
+				// Create ConfigProviderSpec.
+				var configProviderSpec v1alpha1.ConfigProviderSpec
+				err := configProviderSpec.FromKubernetesSecretProviderSpec(k8sSecretConfig)
+				Expect(err).ToNot(HaveOccurred())
+
+				device.Spec.Config = &[]v1alpha1.ConfigProviderSpec{configProviderSpec}
+				logrus.Infof("Updating %s with config %s", deviceId, device.Spec.Config)
+			})
+			Expect(err).ToNot(HaveOccurred())
+
+			logrus.Infof("Waiting for the device to pick the config")
+			harness.WaitForDeviceContents(deviceId, fmt.Sprintf("the device is updated to renderedVersion: %s", strconv.Itoa(newRenderedVersion)),
+				func(device *v1alpha1.Device) bool {
+					for _, condition := range device.Status.Conditions {
+						if condition.Type == "Updating" && condition.Reason == "Updated" && condition.Status == "False" &&
```

Comment:
while i notice this pattern all over the e2e - i think we should be using the proper enums here :
`if condition.Type == v1alpha1.DeviceUpdating && condition.Reason == string(v1alpha1.UpdateStateUpdated) && condition.Status == v1alpha1.ConditionStatusFalse &&.....`
additionally i find it weird that we are checking condition.Reason if its a free text field , if its not it should be backed by its own enum IMO
@eldar101 thoughts?

Reply:
Ok, yeah those are definitely better, some of the existing tests can definitely be refactored to use it.